lxd/firewall/drivers/drivers_util.go

```diff
@@ -36,3 +36,51 @@ func portRangeStr(portRange [2]uint64, delimiter string) string {
 	}
 	return fmt.Sprintf("%d%s%d", portRange[0], delimiter, portRange[0]+portRange[1]-1)
 }
+
+// getOptimisedDNATRanges returns a map of listen port ranges to target port ranges that can be
```

Comment:
I think it could be worth expanding the comment here a bit to explain why this is needed and what it returns in different scenarios to help the future reader understand what the function is doing.